envoy/config/filter/http/jwt_authn/v2/config.proto

```diff
@@ -0,0 +1,207 @@
+
+syntax = "proto3";
+
+package envoy.config.filter.http.jwtauthn.v2;
+
+import "envoy/api/v2/core/http_uri.proto";
+import "envoy/api/v2/route/route.proto";
+import "google/protobuf/duration.proto";
+import "validate/validate.proto";
+
+// JSON Web Token (JWT) token format for authentication is defined
+// `here <https://tools.ietf.org/html/rfc7519>`_.
+// See `OAuth2.0 <https://tools.ietf.org/html/rfc6749>`_ and
+// `OIDC1.0 <http://openid.net/connect>`_ for the authentication flow.
+//
+// Example:
+//
+// .. code-block:: yaml
+//
+//     issuer: https://example.com
+//     audiences:
+//     - bookstore_android.apps.googleusercontent.com
+//       bookstore_web.apps.googleusercontent.com
+//     jwks_uri: https://example.com/.well-known/jwks.json
+//
+// [#not-implemented-hide:]
+message JWT {
+  // Identifies the principal that issued the JWT. See
+  // `here <https://tools.ietf.org/html/rfc7519#section-4.1.1>`_.
+  // Usually a URL or an email address.
+  //
+  // Example: https://securetoken.google.com
+  // Example: 1234567-compute@developer.gserviceaccount.com
+  //
+  string issuer = 1;
+
+  // The list of JWT `audiences
+  // <https://tools.ietf.org/html/rfc7519#section-4.1.3>`_. that are allowed to
+  // access. A JWT containing any of these audiences will be accepted.
+  //
+  // Example:
+  //
+  // .. code-block:: yaml
+  //
+  //     audiences:
+  //     - bookstore_android.apps.googleusercontent.com
+  //       bookstore_web.apps.googleusercontent.com
+  //
+  repeated string audiences = 2;
+
+  // `Json web key set <https://tools.ietf.org/html/rfc7517#appendix-A>`_ URI.
+  // Json web key set is fetched from the URI to validate signature of the JWT.
+  // See `OpenID Discovery
+  // <https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderMetadata>`_
+  //
+  // Example:  jwks_uri:
+  //
+  // .. code-block:: yaml
+  //
+  //    jwks_uri:
+  //    - uri: https://www.googleapis.com/oauth2/v1/certs
+  //      cluster: jwt.www.googleapis.com|443
+  //
+  envoy.api.v2.core.HttpUri jwks_uri = 3;
+
+  // Duration after which the cached public key should be expired. If not
+  // specified, default cache duration is 5 minutes.
+  google.protobuf.Duration public_key_cache_duration = 4;
+
+  // If false, the JWT is removed in the request after a success verification.
+  // Set it true, if don't want it to be removed.
+  // Default value is false which is to remove JWT.
+  bool forward_jwt = 5;
+
+  // If true, the request is allowed if the JWT verification fails.
+  // Default value is false; JWT verification failed request will be rejected.
+  bool allow_failed_jwt = 6;
```

Comment:
So if this is 'false', and validation of the current Jwt fails, you will skip the rest of Jwt rules in the JwtAuthentication? Combination of allow_failed_jwt = false here and the allow_missing_jwt = true seems weird to me in that case.

I think we don't need this (i.e always allow_failed) and control rejection by the boolean in JwtAuthentication only.